java/arcs/core/storage/Refinement.kt

```diff
@@ -11,9 +11,16 @@
 
 package arcs.core.storage
 
-class Refinement<T, Q>(val predicate: (arg: Q, value: T) -> Boolean) {
-    // TODO(cypher1): val toSql: (arg: Q) -> String
-    fun filterBy(arg: Q, values: Set<T>): Set<T> {
-        return values.filter { value -> predicate(arg, value) }.toSet()
+/**
+ * Representation of a refinement type for filtering data from a collection in runtime.
+ *
+ * Supports filtering and data validation via a predicate (optional runtime arguments).
+ */
```

Comment:
An example might be good, or an explanation for what the predicate arguments are (or both).

Reply:
Wrote up a rough explanation with an example use case.